servers/services/src/main/java/org/projectnessie/services/impl/ContentApiImpl.java

```diff
@@ -75,23 +102,49 @@ public ContentResponse getContent(
 
   @Override
   public GetMultipleContentsResponse getMultipleContents(
-      String namedRef, String hashOnRef, List<ContentKey> keys, boolean withDocumentation)
+      String namedRef,
+      String hashOnRef,
+      List<ContentKey> keys,
+      boolean withDocumentation,
+      boolean forWrite)
       throws NessieNotFoundException {
     try {
       ResolvedHash ref =
           getHashResolver()
               .resolveHashOnRef(namedRef, hashOnRef, new HashValidator("Expected hash"));
 
-      BatchAccessChecker check = startAccessCheck().canViewReference(ref.getValue());
+      NamedRef r = ref.getValue();
+      BatchAccessChecker check = startAccessCheck().canViewReference(r);
+      if (forWrite) {
+        check.canCommitChangeAgainstReference(r);
+      }
 
-      Map<ContentKey, ContentResult> values = getStore().getValues(ref.getHash(), keys, false);
+      Map<ContentKey, ContentResult> values = getStore().getValues(ref.getHash(), keys, forWrite);
       List<ContentWithKey> output =
           values.entrySet().stream()
+              .filter(
+                  e -> {
+                    ContentResult contentResult = e.getValue();
+                    IdentifiedContentKey identifiedKey = contentResult.identifiedKey();
+                    check.canReadEntityValue(r, identifiedKey);
+                    if (contentResult.content() != null) {
```

Comment:
You might try to invert these 2 ifs to improve readability.

Reply:
That would add more `if`s